libmultilabel/linear/preprocessor.py

```diff
@@ -1,216 +1,124 @@
 from __future__ import annotations
 
-import csv
 import logging
-import re
-from array import array
-from collections import defaultdict
+import copy
 
 import numpy as np
-import pandas as pd
-import scipy
-import scipy.sparse as sparse
 from sklearn.feature_extraction.text import TfidfVectorizer
 from sklearn.preprocessing import MultiLabelBinarizer
 
-__all__ = ["Preprocessor", "read_libmultilabel_format", "read_libsvm_format"]
+__all__ = ["Preprocessor"]
 
 
 class Preprocessor:
-    """Preprocessor is used to load and preprocess input data in LibSVM and LibMultiLabel formats.
+    """Preprocessor is used to preprocess input data in LibSVM or LibMultiLabel formats.
     The same Preprocessor has to be used for both training and testing data;
-    see save_pipeline and load_pipeline.
+    see save_pipeline and load_pipeline for more details.
     """
 
-    def __init__(self, data_format: str) -> None:
+    def __init__(self, include_test_labels: bool = False, remove_no_label_data: bool = False, tfidf_params: dict = {}):
         """Initializes the preprocessor.
 
         Args:
-            data_format (str): The data format used. 'svm' for LibSVM format, 'txt' for LibMultiLabel format in file and 'dataframe' for LibMultiLabel format in dataframe .
+            include_test_labels (bool, optional): Whether to include labels in the test dataset. Defaults to False.
+            remove_no_label_data (bool, optional): Whether to remove training instances that have no labels. Defaults to False.
+            tfidf_params (dict, optional): Parameters for TfidfVectorizer. Defaults to {}.
+        """
+        self.include_test_labels = include_test_labels
+        self.remove_no_label_data = remove_no_label_data
+        self.tfidf_params = tfidf_params
+        self.data_format = None
+        self.vectorizer = None
+        self.binarizer = None
+
+    def fit(self, dataset):
+        """Fit the raw data.
+
+        Parameters
+        ----------
+        dataset : The training and test dataset along with labels with keys 'train', 'test', and labels respectively.
+            The dataset has keys 'x' for input features and 'y' for labels. It also contains 'data_format' to indicate
+            the data format used.
```

Comment:
The wording does not make it clear these are preconditions.
```suggestion
            The dataset must have keys 'x' for input features, 'y' for labels and 'data_format' to indicate
            the data format used.
```